darwin/exporter/formats/mask.py

```diff
@@ -1,58 +1,390 @@
+    """
+    Returns the colour for the given category name, or generates a new one if it doesn't exist.
+
+    Parameters
+    ----------
+    cat_name: str
+        The name of the category.
+    colours: dt.MaskTypes.ColoursDict
+        A dictionary of category names and their corresponding colours.
+
+    Returns
+    -------
+    int - the integer for the colour name.  These will later be reassigned to a wider spread across the colour spectrum.
+    """
+    if not cat_name in colours:
+        colours[cat_name] = len(colours) + 1
+
+    return colours[cat_name]
+
+
+def render_polygons(
+    mask: NDArray,
+    colours: dt.MaskTypes.ColoursDict,
+    categories: dt.MaskTypes.CategoryList,
+    annotations: List[dt.AnnotationLike],
+    annotation_file: dt.AnnotationFile,
+    height: int,
+    width: int,
+) -> dt.MaskTypes.RendererReturn:
+    """
+    Renders the polygons in the given annotations onto the given mask.
+
+    Parameters
+    ----------
+    mask: NDArray
+        The mask to render the polygons onto.
+    colours: dt.MaskTypes.ColoursDict
+        A dictionary of category names and their corresponding colours.
+    categories: dt.MaskTypes.CategoryList
+        A list of category names.
+    annotations: List[dt.AnnotationLike]
+        A list of annotations to be rendered.
+    annotation_file: dt.AnnotationFile
+        The annotation file that the annotations belong to.
+    height: int
+        The height of the image.
+    width: int
+        The width of the image.
+
+    Returns
+    -------
+    Tuple[List[Exception], Image, dt.MaskTypes.CategoryList, dt.MaskTypes.ColoursDict]
+    """
+
+    errors: List[Exception] = []
+
+    for a in annotations:
+        try:
+            if isinstance(a, dt.VideoAnnotation):
+                print(f"Skipping video annotation from file {annotation_file.filename}")
+                continue
+
+            cat = a.annotation_class.name
+            if not cat in categories:
+                categories.append(cat)
+
+            if a.annotation_class.annotation_type == "polygon":
+                polygon = a.data["path"]
+            elif a.annotation_class.annotation_type == "complex_polygon":
+                polygon = a.data["paths"]
+            else:
+                raise ValueError(f"Unknown annotation type {a.annotation_class.annotation_type}")
+            sequence = convert_polygons_to_sequences(polygon, height=height, width=width)
+
+            colour_to_draw = get_or_generate_colour(cat, colours)
+            mask = draw_polygon(mask, sequence, colour_to_draw)
+
+            if cat not in colours:
+                colours[cat] = colour_to_draw
+
+        except Exception as e:
+            errors.append(e)
+            continue
+
+    # It's not necessary to return the mask, it's modified in place, but it's more explicit
+    return errors, mask, categories, colours
+
+
+def render_raster(
```

Comment:
if wanting the same interface as render_polygon, make sure to note that some inputs aren't used

Reply:
On this one, all imports are used - this is the original interface.